nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -186,7 +186,7 @@ def setup(self, stage=None):
             self.setup_training_dataloader()
         if hasattr(self, '_validation_ds'):
             self._validation_dl = self.setup_eval_dataloader(self._validation_ds, self.cfg.data.validation_ds)
-        if hasattr(self.cfg.data, 'test_ds'):
+        if hasattr(self.cfg.data, 'test_ds') and self.cfg.data.test_ds.get('file_names', None) is not None:
```

Comment:
How come we don't have the same check for validation?

Reply:
Adi says test set should be optional (hence this check) but val set is not optional.